### ISSUE TITLE
IO Worker Panicked - Outputs checked as invariant of "Batch" object

### ISSUE THREAD

**Issue by atlanticcrypto** [paritytech/parity-ethereum#10492]
- **Parity Ethereum version**: 2.3.4
- **Operating system**:  Linux
- **Installation**: binary
- **Fully synchronized**: yes
- **Network**: ethereum

Mining node up since the Constantinople fork gets IO Worker error and dies:

```Mar 19 19:52:45 acc-nexus-2 parity[14646]: 2019-03-19 19:52:45  IO Worker #2 INFO import    435/700 peers    530 MiB chain  141 MiB db  0 bytes queue  346 KiB sync  RPC:  0 conn,   19 req/s,   96 µs
Mar 19 19:53:08 acc-nexus-2 parity[14646]: 2019-03-19 19:53:08  Verifier #16 INFO import  Imported #7402330 0x0964…3953 (38 txs, 1.48 Mgas, 72 ms, 6.80 KiB) + another 1 block(s) containing 56 tx(s)
Mar 19 19:53:12 acc-nexus-2 parity[14646]: ====================
Mar 19 19:53:12 acc-nexus-2 parity[14646]: stack backtrace:
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    0:     0x56075b9e3a7d - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    1:     0x56075b9e2ba2 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    2:     0x56075b9e1d56 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    3:     0x56075a1b1668 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    4:     0x56075ba53e59 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    5:     0x56075ba53901 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    6:     0x56075ba537e5 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    7:     0x56075ba7145c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    8:     0x56075add67d6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    9:     0x56075ae281c5 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   10:     0x56075adc8cdc - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   11:     0x56075adcab5b - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   12:     0x56075ac4ae30 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   13:     0x56075ac352f2 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   14:     0x56075ac7de6b - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   15:     0x56075ac8320c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   16:     0x56075ac3244c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   17:     0x56075ac29252 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   18:     0x56075ac82146 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   19:     0x56075ac29c4a - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   20:     0x56075ac2a0d4 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   21:     0x56075ac2996a - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   22:     0x56075ac29617 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   23:     0x56075ac70715 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   24:     0x56075ac6167c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   25:     0x56075ac6aca6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   26:     0x56075ac5ecc6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   27:     0x56075ba6a909 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   28:     0x56075ac6fa77 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   29:     0x56075ba5e49d - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   30:     0x7f8e6ac536b9 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   31:     0x7f8e6a7733dc - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   32:                0x0 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]: Thread 'IO Worker #3' panicked at 'All outputs checked as invariant of `Batch` object; qed: NoSuchOutput', src/libcore/result.rs:1009
Mar 19 19:53:12 acc-nexus-2 parity[14646]: This is a bug. Please report it at:
Mar 19 19:53:12 acc-nexus-2 parity[14646]:     https://github.com/paritytech/parity-ethereum/issues/new
Mar 19 19:53:12 acc-nexus-2 parity[14646]: 2019-03-19 19:53:12  main INFO parity_ethereum::run  Finishing work, please wait...
Mar 19 19:53:12 acc-nexus-2 parity[14646]: ====================
Mar 19 19:53:12 acc-nexus-2 parity[14646]: stack backtrace:
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    0:     0x56075b9e3a7d - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    1:     0x56075b9e2ba2 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    2:     0x56075b9e1d56 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    3:     0x56075a1b1668 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    4:     0x56075ba53e59 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    5:     0x56075ba53901 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    6:     0x56075ba537e5 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    7:     0x56075ba7145c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    8:     0x56075add67d6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:    9:     0x56075ae281c5 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   10:     0x56075adc8cdc - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   11:     0x56075adcab5b - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   12:     0x56075ac4ae30 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   13:     0x56075ac352f2 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   14:     0x56075ac7de6b - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   15:     0x56075ac8320c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   16:     0x56075ac3244c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   17:     0x56075ac29252 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   18:     0x56075ac82146 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   19:     0x56075ac29c4a - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   20:     0x56075ac2a0d4 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   21:     0x56075ac2996a - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   22:     0x56075ac29617 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   23:     0x56075ac70715 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   24:     0x56075ac6167c - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   25:     0x56075ac6aca6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   26:     0x56075ac5ecc6 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   27:     0x56075ba6a909 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   28:     0x56075ac6fa77 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   29:     0x56075ba5e49d - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   30:     0x7f8e6ac536b9 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   31:     0x7f8e6a7733dc - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]:   32:                0x0 - <no info>
Mar 19 19:53:12 acc-nexus-2 parity[14646]: Thread 'IO Worker #2' panicked at 'All outputs checked as invariant of `Batch` object; qed: NoSuchOutput', src/libcore/result.rs:1009
Mar 19 19:53:12 acc-nexus-2 parity[14646]: This is a bug. Please report it at:
Mar 19 19:53:12 acc-nexus-2 parity[14646]:     https://github.com/paritytech/parity-ethereum/issues/new
Mar 19 19:53:14 acc-nexus-2 parity[14646]: ====================
Mar 19 19:53:14 acc-nexus-2 parity[14646]: stack backtrace:
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    0:     0x56075b9e3a7d - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    1:     0x56075b9e2ba2 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    2:     0x56075b9e1d56 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    3:     0x56075a1b1668 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    4:     0x56075ba53e59 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    5:     0x56075ba53901 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    6:     0x56075ba537e5 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    7:     0x56075ba7145c - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    8:     0x56075add67d6 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:    9:     0x56075ae281c5 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   10:     0x56075adc8cdc - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   11:     0x56075adcab5b - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   12:     0x56075ac4ae30 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   13:     0x56075ac352f2 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   14:     0x56075ac7de6b - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   15:     0x56075ac8320c - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   16:     0x56075ac3244c - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   17:     0x56075ac29252 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   18:     0x56075ac82146 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   19:     0x56075ac29c4a - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   20:     0x56075ac2a0d4 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   21:     0x56075ac2996a - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   22:     0x56075ac29617 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   23:     0x56075ac70715 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   24:     0x56075ac6167c - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   25:     0x56075ac6aca6 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   26:     0x56075ac5ecc6 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   27:     0x56075ba6a909 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   28:     0x56075ac6fa77 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   29:     0x56075ba5e49d - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   30:     0x7f8e6ac536b9 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   31:     0x7f8e6a7733dc - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]:   32:                0x0 - <no info>
Mar 19 19:53:14 acc-nexus-2 parity[14646]: Thread 'IO Worker #0' panicked at 'All outputs checked as invariant of `Batch` object; qed: NoSuchOutput', src/libcore/result.rs:1009
Mar 19 19:53:14 acc-nexus-2 parity[14646]: This is a bug. Please report it at:
Mar 19 19:53:14 acc-nexus-2 parity[14646]:     https://github.com/paritytech/parity-ethereum/issues/new
Mar 19 19:53:15 acc-nexus-2 parity[14646]: ====================
Mar 19 19:53:15 acc-nexus-2 parity[14646]: stack backtrace:
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    0:     0x56075b9e3a7d - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    1:     0x56075b9e2ba2 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    2:     0x56075b9e1d56 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    3:     0x56075a1b1668 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    4:     0x56075ba53e59 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    5:     0x56075ba53901 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    6:     0x56075ba537e5 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    7:     0x56075ba7145c - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    8:     0x56075add67d6 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:    9:     0x56075ae281c5 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   10:     0x56075adc8cdc - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   11:     0x56075adcab5b - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   12:     0x56075ac4ae30 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   13:     0x56075ac352f2 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   14:     0x56075ac7de6b - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   15:     0x56075ac8320c - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   16:     0x56075ac3244c - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   17:     0x56075ac29252 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   18:     0x56075ac82146 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   19:     0x56075ac29c4a - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   20:     0x56075ac2a0d4 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   21:     0x56075ac2996a - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   22:     0x56075ac29617 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   23:     0x56075ac70715 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   24:     0x56075ac6167c - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   25:     0x56075ac6aca6 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   26:     0x56075ac5ecc6 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   27:     0x56075ba6a909 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   28:     0x56075ac6fa77 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   29:     0x56075ba5e49d - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   30:     0x7f8e6ac536b9 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   31:     0x7f8e6a7733dc - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]:   32:                0x0 - <no info>
Mar 19 19:53:15 acc-nexus-2 parity[14646]: Thread 'IO Worker #1' panicked at 'All outputs checked as invariant of `Batch` object; qed: NoSuchOutput', src/libcore/result.rs:1009
Mar 19 19:53:15 acc-nexus-2 parity[14646]: This is a bug. Please report it at:
Mar 19 19:53:15 acc-nexus-2 parity[14646]:     https://github.com/paritytech/parity-ethereum/issues/new
Mar 19 19:54:12 acc-nexus-2 parity[14646]: 2019-03-19 19:54:12  main WARN parity_ethereum::run  Shutdown is taking longer than expected.
Mar 19 19:58:12 acc-nexus-2 parity[14646]: 2019-03-19 19:58:12  main WARN parity_ethereum::run  Shutdown timeout reached, exiting uncleanly.
Mar 19 19:58:13 acc-nexus-2 systemd[1]: parity.service: Main process exited, code=exited, status=1/FAILURE
Mar 19 19:58:13 acc-nexus-2 systemd[1]: parity.service: Unit entered failed state.
Mar 19 19:58:13 acc-nexus-2 systemd[1]: parity.service: Failed with result 'exit-code'.
```

**Comment by schukh** [comment]
+1. One of our node have occurred this problem.

**Comment by Veniamin** [comment]
+1

**Comment by atlanticcrypto** [comment]
This has now happened across multiple nodes. Restarts do not help, reoccurs after about 20 min. Upgrading to 2.3.6 does not help.

**Comment by joshua-mir** [comment]
For everyone seeing this issue, our recommendation right now is to run with `--no-serve-light`.

**Comment by joshua-mir** [comment]
Closed by #10503
